proctor-common/src/main/java/com/indeed/proctor/common/Proctor.java

```diff
@@ -515,4 +523,56 @@ private boolean isIncognitoEnabled(@Nonnull final Map<String, Object> inputConte
                 .map(value -> value.equalsIgnoreCase("true"))
                 .orElse(false);
     }
+
+    /**
+     * Aggregates properties of Payload Experiments for look up in Proctor Result. Checks Priority
+     * of Property to handle conflicting overrides of properties within namespaces. Properties are
+     * the Key:Value pairs of fields stored in JsonNode Payload type.
+     */
+    private void populateProperties(
+            final String testName,
+            final TestBucket testBucket,
+            final Map<String, PayloadProperty> testProperties) {
+        final Payload payload = testBucket.getPayload();
+        if (payload != null && payload.getJson() != null) {
+            payload.getJson()
+                    .fields()
+                    .forEachRemaining(
+                            field -> attemptStoringProperty(field, testName, testProperties));
+        }
+    }
+
+    private void attemptStoringProperty(
+            final Map.Entry<String, com.fasterxml.jackson.databind.JsonNode> field,
+            final String testName,
+            final Map<String, PayloadProperty> testProperties) {
+        final PayloadProperty curr = testProperties.get(field.getKey());
+        // store property if it does not exist in map
+        if (curr == null) {
+            testProperties.put(
+                    field.getKey(),
+                    PayloadProperty.builder().value(field.getValue()).testName(testName).build());
+        } else {
+            final PayloadExperimentConfig currPayloadConfig =
+                    testChoosers
+                            .get(curr.getTestName())
+                            .getTestDefinition()
+                            .getPayloadExperimentConfig();
+            final PayloadExperimentConfig newPayloadConfig =
+                    testChoosers.get(testName).getTestDefinition().getPayloadExperimentConfig();
+            // store property if it has higher priority than the currently stored property
+            try {
+                if (isHigherPriority(currPayloadConfig, newPayloadConfig)) {
```

Comment:
nit: 

it is great that we extracted a method to compare priority.
However it is not clear from reading this line when it returns true, it is the first or second parameter being higher priority(i recognize that my statement can be debatable). 

Will it be better to extract a method call getWinningPropertyValue which will get the highest priority value, which can be existing value or the new value. if the existing value does not exist, then the new value is the winning value. I think it is ok to reinsert the existing value back(to make code shorter) if it is winning again. 

Do we expect priority checking to be evolving? If yes, should we have an interface for priority checker and this class only cares about invoking the interface?

Reply:
I think we can address the interface when we go to do/determine the exact need for the more advanced comparison.

Reply:
Will do the shorten code suggestion

Reply:
I choose to make it a class/instance method so will keep the code as is in respect to shortening the code as no longer applicable.